extensions/qcom/GLSL_QCOM_image_processing2.txt

```diff
@@ -0,0 +1,212 @@
+    | vec4 textureBlockMatchWindowSADQCOM( | Windowed block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSSDQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the squares of the values (SSD).     |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSADQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+
+    * target and reference specify the two input textures. If
+      the block from one texture is going to be used repeatedly across multiple
+      instructions, that texture should be specified as the referenceTexture.
+
+    * targetCoord and referenceCoord specify the texture coordinates (in texels)
+      of the bottom left corner of the block.
+
+    * blockSize specifies the size (in texels) of the blocks to be compared.
+
+
+Restrictions:
+    
+    The <target> and <reference> textures used in textureBlockMatchWindowSADQCOM,
+    textureBlockMatchWindowSSDQCOM, textureBlockMatchGatherSSDQCOM, and/or
+    textureBlockMatchGatherSSDQCOM cannot be used with any other texture functions.
```

Comment:
This should go into the text edits somewhere, otherwise this is likely to be lost or overlooked.

What happens if these restrictions are not followed? Is it the compiler's responsibility to check?

I think this is restricting images that are used with this group of functions from being used with any functions outside the group, rather than images used with any individual function from being used with other image matching functions, but I'm not completely sure.

Reply:
Do you have a good place to which we move the text ?

We ask the table in the extension to be added to the GL spec table in "Texel Loop Functions", and I  found no good place to which to move the "notes" in the section.

I also updated the note to make it match with the note in GLSL_QCOM_image_processing.txt. Let me know if it looks good enough.

Reply:
If it matches the original extension then I think it's fine and this has been waiting on me for quite long, so I'll merge this for you. Sorry it has taken so long. Ideally I think that this text would just go with the function descriptions, possibly in the bullet points that you've added under the table, but consistency is also good.